### PR TITLE
`border0.Listen`, border0 api client lib and examples

### PR DESCRIPTION
**Pull request by waltzofpearls**
This change set includes:

- Border0 API Client SDK
  - Create socket, get socket by name/id and sign socket key
  - Thorough unit test coverage
  - Full godoc coverage
  - Example
- Border Listen SDK
  - Create HTTP listener, which can be used in http.Serve to accept incoming HTTP connections from Border0
  - Full godoc coverage
  - Examples
- Update README
- Github Action for unit tests

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+	"github.com/borderzero/border0-go/client"
+	"github.com/golang-jwt/jwt"
+)
+
+var DefaultClient = client.New(client.WithAPITokenFromEnv())
+
+type Client interface {
+	APITokenClaims() (jwt.MapClaims, error)
+	Request(method, path string, target interface{}, body io.Reader) error
```
[the remark this answers is not in the document]
Comment (waltzofpearls) [reply client.go]: This is not relevant anymore. It was a prototype version, and it has been completely redesigned.

File: client/http_client.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+type HTTPRequester interface {
+	Request(ctx context.Context, method, path string, input, output any) (int, error)
+	Close()
+}
+
+const (
+	headerAccessToken = "x-access-token"
+	headerAccept      = "Accept"
+	headerContentType = "Content-Type"
+	applicationJSON   = "application/json"
```
Comment (adrianosela) [inline client/http_client.go:30]: headerApplicationJSON

File: client/option.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
+package client
+
+import (
+	"time"
+)
+
+// Option is a function that can be passed to NewAPIClient to configure it.
```
Comment (adrianosela) [inline client/option.go:7]: feature request: option to override the underlying http.Client object.

File: client/socket.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
```
Comment (adrianosela) [inline client/socket.go:8]: godoc

File: listen/listener.go [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+	"os"
+	"strings"
+	"time"
+
+	"github.com/borderzero/border0-go/client"
+	"github.com/cenkalti/backoff/v4"
+	"golang.org/x/crypto/ssh"
+)
+
+type Listener struct {
```
Comment (adrianosela) [inline listen/listener.go:26]: godoc

**Review by adrianosela — APPROVED** [review]
approved, add missing godoc 🙏🏻

File: client/option.go (hunk 3 above)
Comment (waltzofpearls) [reply client/option.go:7]: Will be implemented in the next change set.

**Pull request merged**